src/python/WMCore/Database/CMSCouch.py

```diff
@@ -346,6 +346,46 @@ def updateDocument(self, doc_id, design, update_func, fields={}, useBody=False):
                         (self.name, design, update_func, doc_id)
             return self.put(uri=updateUri, data=fields, decode=False)
 
+    def updateBulkDocuments(self, doc_ids, paramsToUpdate, updateLimits=1000):
+
+        uri = '/%s/_bulk_docs/' % self.name
+        conflictDocIDs = []
+        for ids in grouper(doc_ids, updateLimits):
+            # get original documens
+            docs = self.allDocs(options={"include_docs": True}, keys=ids)['rows']
+            data = {}
+            data['docs'] = []
+            for j in docs:
+                doc = {}
+                doc.update(j['doc'])
+                nestedDictUpdate(doc, paramsToUpdate)
+                data['docs'].append(doc)
+
+            if len(data['docs']) > 0:
+                retval = self.post(uri, data)
+                for result in retval:
+                    if result.get('error', None) == 'conflict':
+                        conflictDocIDs.append(result['id'])
+
+        return conflictDocIDs
+
+    def updateBulkDocumentsWithConflictHandle(self, doc_ids, updateParams, updateLimits=1000, maxConflictLimit=10):
+
+        """
+        param: doc_ids: list couch doc ids for updates, shouldn't contain any duplicate or empty string
+        param: updateParams: dictionary of parameters to be updated.
+        param: updateLimits: number of documents in one commit
+        pram maxConflictLimit: number of conflicts to before we give up to fix it to prevent infinite call
+        """
+        conflictDocIDs = self.updateBulkDocuments(doc_ids, updateParams, updateLimits)
+        if len(conflictDocIDs) > 0:
+            # waite a second before trying again for the conflit documents
+            if maxConflictLimit == 0:
+                return conflictDocIDs
+            time.sleep(1)
+            self.updateBulkDocumentsWithConflictHandle(conflictDocIDs, updateParams, maxConflictLimit - 1)
```

Comment:
Third argument has to be a kwd argument (maxConflictLimit=maxConflictLimit - 1). BTW, nice recursive function!

Reply:
good catch, thanks Alan,